src/main/java/picard/sam/util/PrimaryAlignmentKey.java

```diff
@@ -0,0 +1,52 @@
+package picard.sam.util;
+
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * It is useful to define a key such that the key will occur at most once among the primary alignments in a given file
+ * (assuming the file is valid). The read name + pairing status should be sufficient for this.
+ */
+public final class PrimaryAlignmentKey implements Comparable<PrimaryAlignmentKey> {
+
+    private enum PairStatus {UNPAIRED, FIRST, SECOND} // note the order here; it should correspond to that of SAMRecordQueryNameComparator!
```

Comment:
Please put `//` comments before the line they're commenting on, instead of at the end. Also I would say `must` here instead of `should`.

Reply:
Done in previous round.